changefeedccl: add more logging of spans during DistSQL planning

andyyang890 opened the pull request:
This patch adds more logging of spans during various stages of DistSQL
planning to help with debugging plans with erroneous spans.

Epic: None

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/119549)
<!-- Reviewable:end -->

rharding6373 commented:
Would it make sense to open a PR with the extra logging on master and then backport it to 23.1 and 23.2? It seems potentially useful for more than just the 23.1 release.

andyyang890 committed:
changefeedccl: add more logging of spans during DistSQL planning

This patch adds more logging of spans during various stages of DistSQL
planning to help with debugging plans with erroneous spans.

Release note: None

cockroach-teamcity commented:
[![CLA assistant check](https://cla.crdb.dev/pull/badge/not_signed)](https://cla.crdb.dev/cockroachdb/cockroach?pullRequest=119549) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.crdb.dev/cockroachdb/cockroach?pullRequest=119549) before we can accept your contribution.<br/>**13** out of **14** committers have signed the CLA.<br/><br/>:white_check_mark: msbutler<br/>:white_check_mark: rickystewart<br/>:white_check_mark: andyyang890<br/>:white_check_mark: fqazi<br/>:white_check_mark: stevendanna<br/>:white_check_mark: wenyihu6<br/>:white_check_mark: rafiss<br/>:white_check_mark: kvoli<br/>:white_check_mark: pav-kv<br/>:white_check_mark: erikgrinaker<br/>:white_check_mark: Jeremyyang920<br/>:white_check_mark: rail<br/>:white_check_mark: nvanbenschoten<br/>:x: craig[bot]<br/><hr/>**craig[bot]** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.crdb.dev/check/cockroachdb/cockroach?pullRequest=119549) it.</sub>

rharding6373 reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> I only have a couple minor nits. Thanks for adding more observability!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/119549#-:-NrMbP4_-QfdIyCDd7JG:b5d9yb0)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @andyyang890 and @jayshrivastava)

___
*[`pkg/ccl/changefeedccl/changefeed_dist.go` line 399 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/119549#-NrM_GH68GxoJnVtAFHU:-NrM_GH68GxoJnVtAFHV:bdg9q8p) ([raw file](https://github.com/cockroachdb/cockroach/blob/47005956c7dfdd5ac5fb32f2b7faa30aa0a62861/pkg/ccl/changefeedccl/changefeed_dist.go#L399)):*
> ```Go
> 		case rangeDistribution == int64(balancedSimpleDistribution):
> 			if log.ExpensiveLogEnabled(ctx, 2) {
> 				log.Infof(ctx, "attempt to rebalance ranges using balanced simple distribution")
> ```

minor nit: s/attempt to rebalance/rebalancing
___
*[`pkg/ccl/changefeedccl/changefeed_dist.go` line 443 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/119549#-NrM_t8HAs7re7TwDR-b:-NrM_t8HAs7re7TwDR-c:b-id1lzc) ([raw file](https://github.com/cockroachdb/cockroach/blob/47005956c7dfdd5ac5fb32f2b7faa30aa0a62861/pkg/ccl/changefeedccl/changefeed_dist.go#L443)):*
> ```Go
> 		for i, sp := range spanPartitions {
> 			if log.ExpensiveLogEnabled(ctx, 2) {
> 				log.Infof(ctx, "watched spans for node %d: %s", sp.SQLInstanceID, sp)
> ```

This log seems redundant with the previous one. Do you think we need both? I would favor keeping this log in favor of the previous one.


<!-- Sent from Reviewable.io -->

andyyang890 reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/119549)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @jayshrivastava and @rharding6373)

___
*[`pkg/ccl/changefeedccl/changefeed_dist.go` line 399 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/119549#-NrM_GH68GxoJnVtAFHU:-NrMcmTL1KFi3j10dzPn:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/47005956c7dfdd5ac5fb32f2b7faa30aa0a62861/pkg/ccl/changefeedccl/changefeed_dist.go#L399)):*
<details><summary><i>Previously, rharding6373 (Rachael Harding) wrote…</i></summary><blockquote>

minor nit: s/attempt to rebalance/rebalancing
</blockquote></details>

Done.
___
*[`pkg/ccl/changefeedccl/changefeed_dist.go` line 443 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/119549#-NrM_t8HAs7re7TwDR-b:-NrMcnNdAvBAcCPeNfXH:b2ibxck) ([raw file](https://github.com/cockroachdb/cockroach/blob/47005956c7dfdd5ac5fb32f2b7faa30aa0a62861/pkg/ccl/changefeedccl/changefeed_dist.go#L443)):*
<details><summary><i>Previously, rharding6373 (Rachael Harding) wrote…</i></summary><blockquote>

This log seems redundant with the previous one. Do you think we need both? I would favor keeping this log in favor of the previous one.
</blockquote></details>

I think it could still be useful to have both log messages so that we know the spans right after rebalancing (previous log) and the spans right before we pass them to the processors (this log). This way if there are any other immediate transformations, we'll be able to conclusively tell whether rebalancing messed up the spans and/or whether the final spans we're giving to the processors are right. Thoughts?


<!-- Sent from Reviewable.io -->

rharding6373 reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/119549#-:-NrMh4jR5l99Bd-uyXDu:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @jayshrivastava)

___
*[`pkg/ccl/changefeedccl/changefeed_dist.go` line 443 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/119549#-NrM_t8HAs7re7TwDR-b:-NrMgntbEAC8F9zpvlC0:bywx4h9) ([raw file](https://github.com/cockroachdb/cockroach/blob/47005956c7dfdd5ac5fb32f2b7faa30aa0a62861/pkg/ccl/changefeedccl/changefeed_dist.go#L443)):*
<details><summary><i>Previously, andyyang890 (Andy Yang) wrote…</i></summary><blockquote>

I think it could still be useful to have both log messages so that we know the spans right after rebalancing (previous log) and the spans right before we pass them to the processors (this log). This way if there are any other immediate transformations, we'll be able to conclusively tell whether rebalancing messed up the spans and/or whether the final spans we're giving to the processors are right. Thoughts?
</blockquote></details>

Ah, didn't see where they could be modified in between. Seems ok as is.


<!-- Sent from Reviewable.io -->
